modules/alloy/src/main/ts/ephox/alloy/api/ui/FloatingToolbarButton.ts

```diff
@@ -26,12 +26,43 @@ import * as Sketcher from './Sketcher';
 import { Toolbar } from './Toolbar';
 import { CompositeSketchFactory } from './UiSketcher';
 
-const toggle = (button: AlloyComponent, externals: Record<string, any>) => {
+const onClose = (
+  button: AlloyComponent,
+  toolbar: AlloyComponent,
+  ariaControls: AriaControls.AriaManager,
+  options?: { skipFocus: boolean }
+) => {
+  Sandboxing.close(toolbar);
+  if (!options?.skipFocus) {
+    Toggling.off(button);
+    Focusing.focus(button);
+  }
+  ariaControls.unlink(button.element);
+};
+
+const toggle = (
+  button: AlloyComponent,
+  externals: Record<string, any>,
+  detail: FloatingToolbarButtonDetail,
+  spec: FloatingToolbarButtonSpec,
+  ariaControls: AriaControls.AriaManager,
+  options?: { skipFocus: boolean }
+) => {
   const toolbarSandbox = Coupling.getCoupled(button, 'toolbarSandbox');
+
   if (Sandboxing.isOpen(toolbarSandbox)) {
-    Sandboxing.close(toolbarSandbox);
+    onClose(button, toolbarSandbox, ariaControls, options);
   } else {
     Sandboxing.open(toolbarSandbox, externals.toolbar());
+    detail.fetch().get((groups) => {
```

Comment:
What's the main reason you inlined this, rather than just pulling out an `onOpen` to match the `onClose`?

Reply:
The reason for this to get access to the `skipFocus` parameter. I couldn't find a way to get it's value from `onOpen` and `onClose` callbacks.